### PR TITLE
Add predicate methods for previous and next day checks

### PR DESCRIPTION
**Pull request by jatindhankhar**
### Summary

This add supports for checking time against previous day 
`prev_day? `/ `yesterday?`

and next day 

`next_day?` / `tomorrow?`  
#### Intention

While at work, we wanted to check if a date belonged to yesterday, to implement the business logic and were surprised this wasn't part of `activesuppport` 
[Relevant Stackoverflow question](https://stackoverflow.com/questions/26578553/check-if-datetime-value-is-today-tomorrow-or-later)

  
This completes the `today?` predicate with its complimentary helper methods `tomorrow?`, `next_day?`, `yesterday?` and `prev_day?` 

So people can simply use the helper predicates to check it the date in question belongs to the previous or next day.

### CODE REVIEW

**Comment by jatindhankhar** [comment]
Can someone please check this?

**Comment by jatindhankhar** [comment]
cc - @eugeneius
Sorry to bother you, but can you please check this?

**Comment by eugeneius** [comment]
The implementation looks fine. Date is missing test coverage, and maybe the tests could live here instead of being duplicated:
https://github.com/rails/rails/blob/master/activesupport/test/core_ext/date_and_time_behavior.rb

New Active Support core extensions that aren't used internally by Rails are often rejected in favour of being proposed to Ruby directly, but since these complement existing Active Support methods it seems reasonable to consider adding them here. That decision will have to be made by a committer though.

This also needs a changelog entry, and ideally the Active Support Core Extensions guide would be updated to mention the new methods, but I'd recommend waiting for a committer's review before doing more work.

**Comment by jatindhankhar** [comment]
Hi @eugeneius ,
Thanks for the feedback 😄 
Makes sense, tests can be moved to `core_ext/date_and_time_behavior.rb`. 

Yes, I am still waiting for the committer's review (any idea who I can reach out to ? ), saw your comments on come merged PRs with `activesupport` label, so reached out to you.

**Review by jeremy — COMMENTED** [review]
Dig this. Please do update changelog and guides.

**Comment by jatindhankhar** [comment]
> Dig this. Please do update changelog and guides.

Added, please check if this looks fine 😄

File: activesupport/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
+    This adds the predicate helpers that complements `today?`
+
+    Following methods are added
+
+    * `tomorrow?` / `next_day?`  -  Checks if the date/time is of next day (tomorrow)
+    * `yesterday?` /  `prev_day?` -  Checks if the date/time is of previous day (yesterday)
+
+
+    *Jatin Dhankhar*
```
Comment (jeremy) [inline activesupport/CHANGELOG.md]:
```suggestion
*   Add Date and Time `#yesterday?` and `#tomorrow?` alongside `#today?`.

    Aliased to `#prev_day?` and `#next_day?` to match the existing `#prev/next_day` methods.

    *Jatin Dhankhar*
```

File: activesupport/lib/active_support/core_ext/date_and_time/calculations.rb [hunk 2]
```diff
@@ -31,6 +31,18 @@ def today?
       to_date == ::Date.current
     end
 
+    # Returns true if the date/time is of next day (tomorrow)
```
Comment (jeremy) [inline activesupport/lib/active_support/core_ext/date_and_time/calculations.rb]:
```suggestion
    # Returns true if the date/time is tomorrow.
```

File: activesupport/lib/active_support/core_ext/date_and_time/calculations.rb [hunk 3]
```diff
@@ -31,6 +31,18 @@ def today?
...
+    def tomorrow?
+      to_date == ::Date.current.tomorrow
+    end
+    alias_method :next_day?, :tomorrow?
+
+    # Returns true if the date/time is of previous day (yesterday)
```
Comment (jeremy) [inline activesupport/lib/active_support/core_ext/date_and_time/calculations.rb]:
```suggestion
    # Returns true if the date/time is yesterday.
```

**Comment by jatindhankhar** [comment]
Fixed merge conflicts as well.
Thanks for the suggestions :)

File: activesupport/test/core_ext/date_time_ext_test.rb [hunk 4]
```diff
@@ -253,6 +253,78 @@ def test_today_without_offset
...
+      assert_equal true, DateTime.civil(2000, 1, 2, 23, 59, 59, Rational(-18000, 86400)).next_day?
+    end
+  end
+
+  def test_next_day_without_offset
+    Date.stub(:current, Date.new(2000, 1, 1)) do
+      assert_equal false, DateTime.civil(1999, 12, 31, 23, 59, 59).next_day?
+      assert_equal true,  DateTime.civil(2000, 1, 2, 0).next_day?
+      assert_equal false,  DateTime.civil(2000, 1, 1, 23, 59, 59).next_day?
+      assert_equal false, DateTime.civil(2000, 1, 3, 0).next_day?
```
Comment (kamipo) [inline activesupport/test/core_ext/date_time_ext_test.rb]: Can you remove extra spaces or align elements? Almost all extra spaces are looks odd.

File: activesupport/lib/active_support/core_ext/date_and_time/calculations.rb [hunk 5]
```diff
@@ -31,6 +31,18 @@ def today?
       to_date == ::Date.current
     end
 
+    # Returns true if the date/time is tomorrow.
+    def tomorrow?
+      to_date == ::Date.current.tomorrow
+    end
+    alias_method :next_day?, :tomorrow?
```
Comment (kamipo) [inline activesupport/lib/active_support/core_ext/date_and_time/calculations.rb]: Can you use a bit faster `alias` syntax than `alias_method` method for the literal alias?

```suggestion
    alias :next_day :tomorrow?
```

Comment (jatindhankhar) [reply activesupport/lib/active_support/core_ext/date_and_time/calculations.rb]: Sure, replacing all usage of `alias_method`with `alias`

File: activesupport/test/core_ext/date_time_ext_test.rb (hunk 4 above)
Comment (jatindhankhar) [reply activesupport/test/core_ext/date_time_ext_test.rb]: Hi @kamipo 
I see you mean the inconsistent spaces among different assert statements. 
Have aligned them. 
Let me know if it is not up to the mark.

Thanks for reviewing the PR :)

**Pull request merged**